example-external-task-process/src/main/java/info/novatec/micronaut/camunda/externaltask/process/ItemDelegate.java

```diff
@@ -0,0 +1,27 @@
+package info.novatec.micronaut.camunda.externaltask.process;
+
+import org.camunda.bpm.engine.delegate.DelegateExecution;
+import org.camunda.bpm.engine.delegate.JavaDelegate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Singleton;
+import java.util.concurrent.atomic.AtomicInteger;
+
+
+@Singleton
+public class ItemDelegate implements JavaDelegate {
+
+    private static final Logger log = LoggerFactory.getLogger(ItemDelegate.class);
+
+    private AtomicInteger lastCount = new AtomicInteger(0);
+
+    @Override
+    public void execute(DelegateExecution delegateExecution) {
+        final int itemValue = (int) (Math.random() * 1000);
+        delegateExecution.setVariable("itemValue", itemValue);
+        int itemNumber = lastCount.incrementAndGet();
+        delegateExecution.setVariable("itemNumber", itemNumber);
+        log.info("Process Instance for item " + itemNumber + " with Value " + itemValue + " running.");
```

Comment:
besser:
`log.info("Process Instance for item {} with value {} running.", itemNumber, itemValue);`

Reply:
Wieso besser?

Reply:
https://hackernoon.com/you-are-logging-all-wrong-341d4246a6dd

"All three of these versions are inefficient and error prone and should be avoided.

The reason is the call to toString(). it forces the creation of a possibly expensive String object that is then passed to the LogWrapper.debug() method."